exporter/alertmanagerexporter/alertmanager_exporter.go

```diff
@@ -0,0 +1,248 @@
+	settings          component.TelemetrySettings
+	endpoint          string
+	generatorUrl      string
+	defaultSeverity   string
+	severityAttribute string
+}
+
+type alertmanagerEvent struct {
+	spanEvent ptrace.SpanEvent
+	traceID   string
+	spanID    string
+	severity  string
+}
+
+func (s *alertmanagerExporter) convertEventSliceToArray(eventslice ptrace.SpanEventSlice, traceID pcommon.TraceID, spanID pcommon.SpanID) []*alertmanagerEvent {
+	if eventslice.Len() > 0 {
+		events := make([]*alertmanagerEvent, eventslice.Len())
+
+		for i := 0; i < eventslice.Len(); i++ {
+			var severity string
+			// severity := pcommon.NewValueStr(s.defaultSeverity)
+			severityAttrValue, ok := eventslice.At(i).Attributes().Get(s.severityAttribute)
+			if ok {
+				severity = severityAttrValue.AsString()
+			} else {
+				severity = s.defaultSeverity
+			}
+			event := alertmanagerEvent{
+				spanEvent: eventslice.At(i),
+				traceID:   traceID.String(),
+				spanID:    spanID.String(),
+				severity:  severity,
+			}
+
+			events[i] = &event
+		}
+		return events
+	}
+	return nil
+}
+
+func (s *alertmanagerExporter) extractEvents(td ptrace.Traces) []*alertmanagerEvent {
+
+	//Stitch parent trace ID and span ID
+	rss := td.ResourceSpans()
+	var events []*alertmanagerEvent = nil
+	if rss.Len() == 0 {
+		return nil
+	}
+
+	for i := 0; i < rss.Len(); i++ {
+		resource := rss.At(i).Resource()
+		ilss := rss.At(i).ScopeSpans()
+
+		if resource.Attributes().Len() == 0 && ilss.Len() == 0 {
+			return nil
+		}
+
+		for j := 0; j < ilss.Len(); j++ {
+			spans := ilss.At(j).Spans()
+			for k := 0; k < spans.Len(); k++ {
+				traceID := pcommon.TraceID(spans.At(k).TraceID())
+				spanID := pcommon.SpanID(spans.At(k).SpanID())
+				events = append(events, s.convertEventSliceToArray(spans.At(k).Events(), traceID, spanID)...)
+			}
+		}
+	}
+	return events
+}
+
+func createAnnotations(event *alertmanagerEvent) model.LabelSet {
+	LabelMap := make(model.LabelSet, event.spanEvent.Attributes().Len()+1)
+	event.spanEvent.Attributes().Range(func(key string, attr pcommon.Value) bool {
+		LabelMap[model.LabelName(key)] = model.LabelValue(attr.AsString())
+		return true
+	})
+	LabelMap["TraceID"] = model.LabelValue(event.traceID)
+	LabelMap["SpanID"] = model.LabelValue(event.spanID)
+	return LabelMap
+}
+
+func (s *alertmanagerExporter) convertEventstoAlertPayload(events []*alertmanagerEvent) []model.Alert {
+
+	var payload []model.Alert
+	for _, event := range events {
+		annotations := createAnnotations(event)
+
+		alert := model.Alert{
+			StartsAt:     time.Now(),
+			Labels:       model.LabelSet{"severity": model.LabelValue(event.severity), "event_name": model.LabelValue(event.spanEvent.Name())},
+			Annotations:  annotations,
+			GeneratorURL: s.generatorUrl,
+		}
+
+		payload = append(payload, alert)
+	}
+	return payload
+}
+
+func (s *alertmanagerExporter) postAlert(ctx context.Context, payload []model.Alert) error {
+	msg, _ := json.Marshal(payload)
```

Comment:
This is swallowing a potential error. It would be best to handle it accordingly.